### PR TITLE
Intensity rotation

### PR DESCRIPTION
**Pull request by gpospelov**
Rotation of intensity data will also remove masks, if exist

### CODE REVIEW

File: GUI/coregui/Views/MaskWidgets/MaskEditorCanvas.cpp [hunk 1]
```diff
@@ -34,6 +34,18 @@
 #include <QModelIndex>
 #include <QVBoxLayout>
 
+namespace {
+inline bool getRotateWarningCallback(QWidget* parent) {
+    const QString title("Rotate data");
+
+    const QString message("Rotation will break the link between the data and the instrument. "
+                          "Detector masks, if exist, will be removed.");
+
```
Comment (waltervh) [inline GUI/coregui/Views/MaskWidgets/MaskEditorCanvas.cpp]: "if exist" -> "if they exist"

File: GUI/coregui/Views/MaskWidgets/MaskEditorCanvas.cpp [hunk 2]
```diff
@@ -35,15 +35,40 @@
...
+bool isRotationAffectsSetup(IntensityDataItem& intensityItem) {
+    if (intensityItem.parent()->getItemValue(RealDataItem::P_INSTRUMENT_ID).toBool())
+        return true;
+
+    if (intensityItem.maskContainerItem() && intensityItem.maskContainerItem()->hasChildren())
+        return true;
+
+    return false;
+}
+
```
Comment (waltervh) [inline GUI/coregui/Views/MaskWidgets/MaskEditorCanvas.cpp]: Rename: isRotationAffectsSetup -> rotationAffectsSetup

File: GUI/coregui/Views/MaskWidgets/MaskEditorCanvas.cpp [hunk 3]
```diff
@@ -35,7 +35,7 @@
 #include <QVBoxLayout>
 
 namespace {
-bool getRotateWarningCallback(QWidget* parent) {
+bool riseRotateWarningDialog(QWidget* parent) {
     const QString title("Rotate data");
```
Comment (waltervh) [inline GUI/coregui/Views/MaskWidgets/MaskEditorCanvas.cpp]: Wrong formulation: a warning can be 'raised', a dialog is just 'opened', so "openRotateWarningDialog"

File: GUI/coregui/Views/MaskWidgets/MaskEditorCanvas.cpp [hunk 4]
```diff
@@ -35,15 +35,40 @@
...
 
-    return GUIHelpers::question(parent, title, message, "Do you wish to roate the data?",
+    return GUIHelpers::question(parent, title, message, "Do you wish to rotate the data?",
         "Yes, please rotate", "No, cancel data rotation");
 }
+
+//! Returns true, if rotation will affect linked instrument or mask presence.
+
+bool isRotationAffectsSetup(IntensityDataItem& intensityItem) {
+    if (intensityItem.parent()->getItemValue(RealDataItem::P_INSTRUMENT_ID).toBool())
```
Comment (DmitryYurov) [inline GUI/coregui/Views/MaskWidgets/MaskEditorCanvas.cpp]: if (intensityItem.parent() && intensityItem.parent()->getItemValue(RealDataItem::P_INSTRUMENT_ID).toBool()) could be a safer solution

Comment (gpospelov) [reply GUI/coregui/Views/MaskWidgets/MaskEditorCanvas.cpp]: In principle yes, but all our items have parent, except very top one, the root...

**Pull request merged**